### DISCUSSION TITLE
Password encryption

### DISCUSSION THREAD

**Discussion by mrvillage** [supabase/supabase#1942] [Questions] [answered]
Hello! I'm hoping to be able to manually set and change passwords but I'm not sure how to do the encryption step. Is there some way to read/write passwords to the database with the encryption Supabase Auth is expecting for the signIn endpoint? Thanks!

Edit: Or simply some way I can change the password of a user in a server-side manner :)

**Comment by easy-street** [comment, marked as answer]
you can change the password of a user server-side with [`auth.update()`](https://supabase.io/docs/reference/javascript/auth-update):

```
supabase.auth.setAuth(token); // token is the user's JWT access token

const { data, error } = await supabase.auth.update({ password: "secret" });
```

**Reply by mrvillage** [reply]
Oh nice! Is there a way to generate an access token server-side though? With my service key maybe?

**Reply by easy-street** [reply]
in a web app, the user's access token is typically extracted from a header or cookie. the [nextjs-with-supabase-auth example](https://github.com/supabase/supabase/blob/master/examples/nextjs-with-supabase-auth/pages/index.js#L23) is a reference implementation for a cookie-based approach.

for a standalone service, it's certainly possible to mint JWTs with the `jwt_secret` from the dashboard. some context on this:
- https://github.com/supabase/supabase/discussions/1849#discussioncomment-837636
- https://supabase.io/docs/learn/auth-deep-dive/auth-deep-dive-jwts#what-are-json-web-tokens-jwts

**Reply by mrvillage** [reply]
Oh awesome!!! Thanks a lot!
